internal/clients/team4/speaker.go

```diff
@@ -11,21 +11,67 @@ type speaker struct {
 	parent *client
 }
 
-//s.parent.ServerReadHandle.GetGameConfig()
+//SpeakerActionPriorities indicate speaker actions in order of priority
+var SpeakerActionPriorities = []string{
+	"SetVotingResult",
+	"SetRuleToVote",
+	"AnnounceVotingResult",
+	"UpdateRules",
+	"AppointNextJudge",
+}
+
+func (s *speaker) GetSpeakerBudget() shared.Resources {
+	return s.parent.ServerReadHandle.GetGameState().IIGORolesBudget[shared.Speaker]
+}
+
+func (s *speaker) GetActionsCost(actions []string) shared.Resources {
+	gameconfig := s.parent.ServerReadHandle.GetGameConfig().IIGOClientConfig
+	costs := map[string]shared.Resources{
+		"SetVotingResult":      gameconfig.SetVotingResultActionCost,
+		"SetRuleToVote":        gameconfig.SetRuleToVoteActionCost,
+		"AnnounceVotingResult": gameconfig.AnnounceVotingResultActionCost,
+		"UpdateRules":          gameconfig.UpdateRulesActionCost,
+		"AppointNextJudge":     gameconfig.AppointNextJudgeActionCost,
+	}
+	var SumOfCosts shared.Resources = 0
+	for _, action := range actions {
+		SumOfCosts += costs[action]
+	}
+	return SumOfCosts
+}
+
+func (s *speaker) GetHigherPriorityActionsCost(baseaction string) shared.Resources {
+	actionindex := len(SpeakerActionPriorities)
+	for i, action := range SpeakerActionPriorities {
+		if action == baseaction {
+			actionindex = i
+		}
+	}
+	return s.GetActionsCost(SpeakerActionPriorities[:actionindex])
+}
 
 // PayJudge is used for paying judge for his service
-/*func (s *speaker) PayJudge() shared.SpeakerReturnContent {
+func (s *speaker) PayJudge() shared.SpeakerReturnContent {
 	JudgeSalaryRule, ok := rules.RulesInPlay["salary_cycle_judge"]
 	var JudgeSalary shared.Resources = 0
```

Comment:
You always start by assigning a value, so the initialisation is not strictly necessary.
Also perhaps move declarations to the top?

Reply:
Minor comment. See the comment bellow first

Reply:
> You always start by assigning a value, so the initialisation is not strictly necessary.
> Also perhaps move declarations to the top?

That's only if the if statements are true. If none of the if statements are true, the function returns an inexistent value.

Reply:
Sorry, my mistake again